ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -580,7 +620,17 @@ let dss_mem_vms doms =
       in
       let other_ds =
         if domid = 0 then
-          None
+          match mem_available () with
+          | Ok mem ->
+              Some
+                ( Rrd.VM uuid
+                , Ds.ds_make ~name:"memory_internal_free" ~units:"B"
+                    ~description:"Dom0 current free memory"
+                    ~value:(Rrd.VT_Int64 mem) ~ty:Rrd.Gauge ~min:0.0
+                    ~default:true ()
+                )
+          | Error _ ->
```

Comment:
Any error should be logged. The error to log is in this value.

Reply:
error log was added.